### PR TITLE
Legend types and maxHeight

### PR DESCRIPTION
**Pull request by mbarrenechea**
### Overview

This PR adds:
- Legend types

  - Choropleth
  - Basic
  - Gradient

- Max height

### CODE REVIEW

File: app/components/map/legend/component.stories.tsx [hunk 1]
```diff
@@ -27,14 +32,22 @@ const Template: Story<LegendProps> = (args) => {
...
+          <LegendItem
+            key={i.id}
+            {...i}
+          >
+            {type === 'basic' && <LegendTypeBasic className="text-sm text-gray-300" items={items} />}
+            {type === 'choropleth' && <LegendTypeChoropleth className="text-sm text-gray-300" items={items} />}
+            {type === 'gradient' && <LegendTypeGradient className="text-sm text-gray-300" items={items} />}
+          </LegendItem>
+        );
+      })}
```
Comment (kgajowy) [inline app/components/map/legend/component.stories.tsx:50]:
```
// pseudocode
{sortedItems.map(({type, items, id, ...rest}) => <LegendItem
            key={i.id} // wouldn't  type be better?
            {...rest}
          >
            {type === 'basic' && <LegendTypeBasic className="text-sm text-gray-300" items={items} />}
            {type === 'choropleth' && <LegendTypeChoropleth className="text-sm text-gray-300" items={items} />}
            {type === 'gradient' && <LegendTypeGradient className="text-sm text-gray-300" items={items} />}
          </LegendItem>
)}
```

File: app/components/map/legend/types/gradient/component.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import cx from 'classnames';
+
+export interface LegendTypeGradientProps {
+  className?: string;
+  items: Array<{
+    value: string;
+    color: string;
+  }>;
+}
```
Comment (kgajowy) [inline app/components/map/legend/types/gradient/component.tsx:10]: Any reason why legends cannot share the same interface?

File: app/components/map/legend/component.stories.tsx (hunk 1 above)
Comment (mbarrenechea) [reply app/components/map/legend/component.stories.tsx:50]: _**"wouldn't  type be better?"**_
Not at all. If you have legend items of the same type you will have the same keys for different items. Not desirable

File: app/components/map/legend/types/gradient/component.tsx (hunk 2 above)
Comment (mbarrenechea) [reply app/components/map/legend/types/gradient/component.tsx:10]: It depends on the type. It's true that now they have the same interface, but I could imagine basic type having their own items. For example, you could define the icon for basic ones. That's why I want to keep them separate.

**Comment by mbarrenechea** [comment]
@kgajowy thanks for reviewing!!

File: app/components/map/legend/component.stories.tsx (hunk 1 above)
Comment (kgajowy) [reply app/components/map/legend/component.stories.tsx:50]: Ah indeed, I wrongly assumed that only those 3 will be present (i.e. it couldn't have multiple `basic` types)

File: app/components/map/legend/types/gradient/component.tsx (hunk 2 above)
Comment (kgajowy) [reply app/components/map/legend/types/gradient/component.tsx:10]: @mbarrenechea sure thing, they can just extend base later on - but they still have the same base ; ) Even tho they have different name, TS consider them to be exactly the same.

**Pull request merged**